sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java

```diff
@@ -105,9 +105,9 @@ private static Object getValueByColumnType(final ResultSet resultSet, final int
             case Types.TINYINT:
             case Types.SMALLINT:
             case Types.INTEGER:
-                return resultSet.getInt(columnIndex);
+                return getIntTypeValue(resultSet, columnIndex);
```

Comment:
TINYINT and SAMLLINT use getInt is enough

INTEGER if isSigned, should still getInt, if isUnsigned can getLong.